llvm/test/CodeGen/RISCV/O3-pipeline.ll

```diff
@@ -115,6 +115,7 @@
 ; CHECK-NEXT:       MachinePostDominator Tree Construction
 ; CHECK-NEXT:       Machine Cycle Info Analysis
 ; CHECK-NEXT:       Machine code sinking
+; CHECK-NEXT:       MachineDominator Tree Construction
```

Comment:
This is an expensive change for compile time.

Reply:
So do you want me to revert it and keep the two functions separate?

Reply:
I think I resolved it. Pushing now.

Reply:
I'm quite confused why its running `MachineDominator Tree` **before** the pass that now invalidates it?

Reply:
No longer an issue. I made a change that keeps it as it was.